### PR TITLE
[Quantum RPG] Add engineer to the party.

### PR DESCRIPTION
**Pull request by dstrain115**
In the NMR lab, talking to the engineer adds the
qaracter to the party.

### CODE REVIEW

File: unitary/examples/quantum_rpg/final_state_preparation/classical_realm.py [hunk 1]
```diff
@@ -50,6 +52,38 @@ def _green_foam(number: int, prob: float = 0.5, xp=_GREEN_XP):
...
 
+def _engineer_joins(state: GameState) -> str:
+    if len(state.party) > 1:
+        return f"The engineer reminisces about his former experiment."
+    print("The engineer looks at the apparatus that dominates the room.")
+    print("'NMR has been a promising technology for quantum computing,'")
+    print("the engineer says.  'And we have learned a lot.  But, ultimately,")
+    print("I can already see that this technology is not scalable and will")
+    print("not solve the problems that plague us today.  I will leave this")
+    print("for the scientists wotking on other technologies and join you")
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/classical_realm.py:63]: wotking 
working

File: unitary/examples/quantum_rpg/final_state_preparation/classical_realm.py [hunk 2]
```diff
@@ -50,6 +52,38 @@ def _green_foam(number: int, prob: float = 0.5, xp=_GREEN_XP):
...
+    print("on the journey towards quantum error correction.!")
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/classical_realm.py:64]: .!
!

File: unitary/examples/quantum_rpg/final_state_preparation/final_state_world_test.py [hunk 3]
```diff
@@ -65,3 +68,14 @@ def test_consistent_exits():
 def test_classical_realm_paths(path, expected_location):
     test_world = go_directions(path)
     assert test_world.current_location.label == expected_location
+
+
+def test_engineer():
+    c = classes.Analyst("Mensing")
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/final_state_world_test.py:74]: I think I missed a step. You seem to create an Analyst, but you're writing a test for Engineer. Where is Engineer created?

Is the engineer (named Hamilton) created and added to the party when a character (Mensing the Analyst, in this case) visits the right room?

**Review by cognigami — CHANGES_REQUESTED** [review]

File: unitary/examples/quantum_rpg/final_state_preparation/final_state_world_test.py (hunk 3 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/final_state_preparation/final_state_world_test.py:78]: Yes, that's correct.  If you talk to the engineer (who is in the NMR lab), the engineer will join the party.  This tests this functionality.  Added some comments to the code to clarify.

File: unitary/examples/quantum_rpg/final_state_preparation/classical_realm.py (hunk 1 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/final_state_preparation/classical_realm.py]: Done.

File: unitary/examples/quantum_rpg/final_state_preparation/classical_realm.py (hunk 2 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/final_state_preparation/classical_realm.py]: Done.

**Review by cognigami — APPROVED** [review]

**Pull request merged**